hi-ml-cpath/src/health_cpath/datamodules/base_module.py

```diff
@@ -276,79 +274,40 @@ class SlidesDataModule(HistoDataModule[SlidesDataset]):
-        tile_count, then sort by intensity sum, and take the smallest (for min), largest (for max) or random (for
-        random) subset, defaults to "min" (which assumes background is high value). This param is passed to TileOnGridd
-        monai transform for tiling on the fly.
-        :param backend: the WSI reader backend, defaults to "cuCIM". This param is passed to LoadImaged monai transform
-        :param wsi_reader_args: Additional arguments to pass to the WSIReader, defaults to {}. Multi processing is
+        :param backend: the WSI reader backend, defaults to "cuCIM".
+        :param wsi_reader_args: additional arguments to pass to the WSIReader, defaults to {}. Multi processing is
         enabled since monai 1.0.0 by specifying num_workers > 0 with CuCIM backend only.
+        :param tiling_params: the tiling on the fly parameters, defaults to TileOnTheFlyParams()
         """
         super().__init__(**kwargs)
-        # Tiling on the fly params
-        self.tile_size = tile_size
-        self.step = step
-        self.random_offset = random_offset
-        self.pad_full = pad_full
-        self.background_val = background_val
-        self.filter_mode = filter_mode
+        self.tiling_params = tiling_params
         # WSIReader params
         self.level = level
         self.backend = backend
         self.wsi_reader_args = wsi_reader_args
-        # TileOnGridd transform expects None to select all foreground tile so we hardcode max_bag_size and
-        # max_bag_size_inf to None if set to 0
-        for stage_key, max_bag_size in self.bag_sizes.items():
-            if max_bag_size == 0:
-                self.bag_sizes[stage_key] = None  # type: ignore
 
     def _load_dataset(self, slides_dataset: SlidesDataset, stage: ModelKey) -> Dataset:
         base_transform = Compose(
             [
                 LoadImaged(
-                    keys=slides_dataset.IMAGE_COLUMN,
+                    keys=SlideKey.IMAGE,
                     reader=WSIReader,
                     dtype=np.uint8,
                     image_only=True,
                     level=self.level,
                     backend=self.backend,
                     **self.wsi_reader_args,
                 ),
-                TileOnGridd(
-                    keys=slides_dataset.IMAGE_COLUMN,
-                    tile_count=self.bag_sizes[stage],
-                    tile_size=self.tile_size,
-                    step=self.step,
-                    random_offset=self.random_offset if stage == ModelKey.TRAIN else False,
-                    pad_full=self.pad_full,
-                    background_val=self.background_val,
-                    filter_mode=self.filter_mode,
-                    return_list_of_dicts=True,
-                ),
+                self.tiling_params.get_tiling_transform(bag_size=self.bag_sizes[stage], stage=stage),
+                SplitDimd(keys=SlideKey.IMAGE, dim=0, keepdim=False, list_output=True),
```

Comment:
What does this `SplitDimd` transform do here? Why is it needed? Maybe helpful to add an explanatory comment.

Reply:
GridPatch returns the tiles already stacked, we need to slipt them to be able to apply other augmentations that need to be applied on a single tile like geometric transforms... will Add a comment